app/src/containers/b2b/Dashboard.tsx

```diff
@@ -0,0 +1,404 @@
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: true,
+      noSearchResults: false,
+      showSearchLoader: false,
+      defaultBillingAddress: {
+        name: 'Inez Larson',
+        address: '198 Bendar Knoll',
+        city: 'East Nicklaus',
+        state: 'Michigan',
+        zip: '48002',
+        country: 'United States',
+      },
+      defaultShippingAddress: {
+        name: 'Max Wilkerson',
+        address: '198 Bendar Knoll',
+        city: 'East Nicklaus',
+        state: 'Michigan',
+        zip: '48002',
+        country: 'United States',
+      },
+      recentOrders: [
+        {
+          orderId: '170-05-3731',
+          date: '03 May 2019',
+          shipTo: 'Max Wilkerson',
+          orderTotal: '$4628.77',
+          status: 'Complete',
+        },
+        {
+          orderId: '170-05-3730',
+          date: '01 May 2019',
+          shipTo: 'Max Wilkerson',
+          orderTotal: '$4308.32',
+          status: 'Complete',
+        },
+        {
+          orderId: '170-05-3729',
+          date: '27 Apr 2019',
+          shipTo: 'Max Wilkerson',
+          orderTotal: '$3182.27',
+          status: 'Processing',
+        },
+        {
+          orderId: '170-05-3728',
+          date: '26 Apr 2019',
+          shipTo: 'Christopher Bryan',
+          orderTotal: '$2934.03',
+          status: 'Processing',
+        },
+      ],
+      accounts: [],
+      admins: [],
+      searchAccounts: '',
+    };
+    this.getAdminData();
+    this.setSearchAccounts = this.setSearchAccounts.bind(this);
+    this.getSearchAccounts = this.getSearchAccounts.bind(this);
+    this.handleEnterKeyPress = this.handleEnterKeyPress.bind(this);
+  }
+
+  setSearchAccounts(event) {
+    this.setState({ searchAccounts: event.target.value });
+  }
+
+  getAdminData() {
+    login().then(() => {
+      adminFetch(`/?zoom=${accountsZoomArray.join()}`, {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthTokenAuthService`),
+        }
+      })
+        .then(res => res.json())
+        .then(res => {
+          if (res && res._accounts) {
+            const accounts = res._accounts[0]._element.map(account => {
+              const uri = account.self.uri.split('/').pop();
+              return {
+                name: account.name,
+                externalId: account['external-id'],
+                status: account._statusinfo[0]._status[0].status.toLowerCase(),
+                uri
+              }
+            });
+            let map = new Map();
+            res._accounts[0]._element.reduce((accum, account) => {
+              const associates = account._associateroleassignments[0]._element;
+              if(!associates) return accum;
+
+              associates.forEach(associate => {
+                if (associate._roleinfo[0]._roles[0]._element[0].name === 'BUYER_ADMIN') {
```

Comment:
This can be null if there are no roles assigned to the associate. Need to break this down into smaller checks:
![image](https://user-images.githubusercontent.com/39800563/61297139-94d7e800-a7a9-11e9-87f5-8b7333856d22.png)